3d/mmd_pmd.ksy

```diff
@@ -0,0 +1,371 @@
+      - id: face_vertex_count
+        type: u4
+      - id: file_name
+        type: strz
+        size: 20
+
+  bone:
+    seq:
+      - id: name
+        type: strz
+        size: 20
+      - id: parent_index
+        type: u2
+      - id: tail_index
+        type: u2
+      - id: type
+        type: u1
+      - id: ik_index
+        type: u2
+      - id: position
+        type: vec3
+
+  ik:
+    seq:
+      - id: target
+        type: u2
+      - id: effector
+        type: u2
+      - id: link_count
+        type: u1
+      - id: iteration
+        type: u2
+      - id: max_angle
+        type: f4
+      - id: link_indices
+        type: u2
+        repeat: expr
+        repeat-expr: link_count
+
+  morph:
+    seq:
+      - id: name
+        type: strz
+        size: 20
+      - id: element_count
+        type: u4
+      - id: type
+        type: u1
+      - id: elements
+        type: morph_element
+        repeat: expr
+        repeat-expr: element_count
+
+  morph_element:
+    seq:
+      - id: index
+        type: u4
+      - id: position
+        type: vec3
+
+  morph_frame:
+    seq:
+      - id: index
+        type: u2
+
+  bone_frame_name:
+    seq:
+      - id: name
+        type: strz
+        size: 50
+
+  bone_frame:
+    seq:
+      - id: bone_index
+        type: u2
+      - id: frame_index
+        type: u1
+
+  english_header:
+    seq:
+      - id: compatibility
+        type: u1
+      - id: model_name
+        type: strz
+        size: 20
+        if: compatibility > 0
+      - id: comment
+        type: strz
+        size: 256
+        if: compatibility > 0
+
+  english_bone_name:
+    seq:
+      - id: name
+        type: strz
+        size: 20
+
+  english_morph_name:
+    seq:
+      - id: name
+        type: strz
+        size: 20
+
+  english_bone_frame_name:
+    seq:
+      - id: name
+        type: strz
+        size: 50
+
+  toon_texture:
+    seq:
+      - id: file_name
+        type: strz
+        size: 100
+
+  rigid_body:
+    seq:
+      - id: name
+        type: strz
+        size: 20
+      - id: bone_index
+        type: u2
+      - id: group_index
+        type: u1
+      - id: group_target
+        type: u2
+      - id: shape_type
+        type: u1
+      - id: width
+        type: f4
+      - id: height
+        type: f4
+      - id: depth
+        type: f4
+      - id: position
+        type: vec3
+      - id: rotation
+        type: vec3
+      - id: weight
+        type: f4
+      - id: position_damping
+        type: f4
+      - id: rotation_damping
+        type: f4
+      - id: restitution
+        type: f4
+      - id: friction
+        type: f4
+      - id: type
+        type: u1
+
+  constraint:
+    seq:
+      - id: name
+        type: strz
+        size: 20
+      - id: rigid_body_index1
+        type: u4
+      - id: rigid_body_index2
+        type: u4
+      - id: position
+        type: vec3
+      - id: rotation
+        type: vec3
+      - id: translation_limitation1
```

Comment:
These on the other hand I would leave as-is because I strongly suspect that it's a lower/upper pair and that naming would make more sense. To rename it, I would have to figure out which is which, though.